test/browser/keys.test.js

```diff
@@ -372,21 +372,21 @@ describe('keys', () => {
 		ops = [];
 		render(<Foo moved={false} />, scratch);
 
-		expect(ops).to.deep.equal(['Mount Stateful2', 'Mount Stateful1']);
+		expect(ops).to.deep.equal(['Mount Stateful1', 'Mount Stateful2']);
```

Comment:
Previously Preact called `cDM` for siblings in reverse order. Now we call `cDM` for siblings from top to bottom. Since `cDM` is invoked after all DOM operations are done, I don't think this order matters as much, but for what its worth, our order matches React now ([codesandbox](https://codesandbox.io/s/react-lifecycle-render-callback-ordering-rmvwp)).

This sibling re-ordering also impacted the `should handle hoisted component vnodes without DOM` test.